include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -119,6 +124,31 @@ class MotorInterruptHandler {
         }
         return false;
     }
+    void send_ack(bool lim_sw_status, AckMessageId ack_msg_id) {
+        if (buffered_move.group_id != NO_GROUP) {
+            auto ack = Ack{.group_id = buffered_move.group_id,
+                           .seq_id = buffered_move.seq_id,
+                           .current_position = static_cast<uint32_t>(
+                               position_tracker >>
+                               31),  // TODO (AA 2021-11-10): convert
+                                     // this value to mm instead of steps
+                           .ack_id = ack_msg_id,
+                           .lim_sw_triggered = lim_sw_status};
+            static_cast<void>(
+                status_queue_client.send_move_status_reporter_queue(ack));
+        }
+    }
+    auto limit_switch_triggered() -> bool {
+        if (hardware.check_limit_switch() && can_step()) {
+            send_ack(true, AckMessageId::error);
+            return true;
+        }
+        if (hardware.check_limit_switch() && !can_step()) {
+            send_ack(true, AckMessageId::complete);
```

Comment:
Now that I think about it, we want to have the move be "finished" regardless of what happens in the limit switch.


We should be calling `finish_move` either here or higher up in the `pulse` function